### DISCUSSION TITLE
Skip Link

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#252] [Component suggestions]
<img width="480" alt="visuele weergave van het skip link component" src="https://github.com/nl-design-system/backlog/assets/545207/f818c64d-7543-498f-8e00-eaf121b70c36">

We willen graag meer leren over het Skip Link component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1715680682790989)

---

## Naam
Skip Link

## Doel
Verborgen mogelijkheid om snel naar een ander deel van de pagina te navigeren zonder gebruik te maken van een muis of touchscreen.

## Richtlijnen

Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Skip Link](https://nldesignsystem.nl/skip-link)

## Onderzoek

[Skip Link component - GOV.UK](https://design-system.service.gov.uk/components/skip-link/)
Onder 'When to use this component' wordt de positie van de Skip Link in combinatie met een cookie melding benoemd maar ook dat een Skip Link niet in een `nav` element zou moeten staan.

[Skip Link Github Discussion -  GOV.UK](https://github.com/alphagov/govuk-design-system-backlog/issues/66)
In deze Discussion van GOV.UK worden enkele toepassingen besproken zoals de Skip Link in combinatie met de Back link maar ook hoe de Skip Link kan helpen om een filter over te slaan.

[Skip Navigation Links - WebAIM](https://webaim.org/techniques/skipnav/)
Artikel bij WebAIM over het gebruik en toepassen van Skip Links.

[scroll-margin-top - MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-margin-top)
CSS techniek die kan helpen om iets meer margin-top mee te geven aan het onderdeel waar de Skip Link naartoe linkt.

## Richtlijnen
- https://nldesignsystem.nl/wcag/2.4.1/
- https://www.w3.org/WAI/WCAG22/Understanding/bypass-blocks.html

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-navigation-skip-link--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/skip-link)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=9679-11007&t=BCAjTQpVcRMAOo7b-4)
- Skip link toegepast op https://www.amsterdam.nl

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-05-14 om 12 53 51](https://github.com/nl-design-system/backlog/assets/545207/4070ebc9-754b-45d5-beb4-a0ce90853f4d)


**Figma**
![Scherm­afbeelding 2024-05-14 om 12 55 39](https://github.com/nl-design-system/backlog/assets/545207/f90f17be-d834-4c27-add7-0e7c601b17cd)


**Website**
![Scherm­afbeelding 2024-05-14 om 12 59 58](https://github.com/nl-design-system/backlog/assets/545207/9e3b5b21-3659-450c-9910-733bec4d51d7)


---

### Notities
- Richtlijnen beschikbaar (Storybook).
- Positie: bovenaan, drukt content naar beneden.
- Visual design: Gecentreerde witte tekst op donkerblauw vlak die over de gehele breedte uitvult. Skip link op [amsterdam.nl](https://www.amsterdam.nl/) wijkt qua vorm (nog) af.
- Label: 'Direct naar inhoud'.
- Focus:
  - In de documentatie (Storybook) wordt aangegeven dat de focus bij een artikelpagina bijvoorbeeld naar de titel van het artikel kan verwijzen en op een zoekpagina naar het zoekveld.
  - Op [homepage](https://www.amsterdam.nl/) verwijst de Skip link naar een `div` met `id="Content"`, deze bevindt zich direct onder de Page Header (inclusief navigatie).
  - Op een [detailpagina](https://www.amsterdam.nl/burgerzaken/rijbewijs/eerste-rijbewijs-aanvragen/) en [zoekpagina](https://www.amsterdam.nl/zoeken/?Zoe=ajax&current_page=1) verwijst de Skip link naar een `div` met `id="Content"`, deze bevindt zich direct onder de Breadcrumbs.
- De Skip link op de website staat in een `<li>` en heeft de `class="skiplinks"`, wat doet vermoeden dat er meer links aan toegevoegd kunnen worden.
- De Skip link gaat vooraf aan de `div` waar de inhoud van de Page header in zit.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-skiplink--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/SkipLink)
- [Figma - WIP](https://www.figma.com/design/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?node-id=2217-10123&t=ucV0dMkg06ub1FpZ-4)
- Skip link wordt toegepast op https://www.denhaag.nl/

---

### Screenshots / Varianten

**Figma - WIP**
![Scherm­afbeelding 2024-05-14 om 15 51 24](https://github.com/nl-design-system/backlog/assets/545207/d3acf611-d32d-4577-9f2f-8466159682a9)

**Website**
![Scherm­afbeelding 2024-05-14 om 15 51 36](https://github.com/nl-design-system/backlog/assets/545207/07dee6c7-cc3a-4c1d-a693-297a18f5ffec)

---

### Notities

Website
- Positie: Linksboven, over de content heen.
- Visual design: Donkere tekst op lichtgeel vlak, breedte op basis van tekst.
- Label: 'Ga naar hoofdinhoud'
- Focus:
  - Op [homepage](https://www.denhaag.nl/) verwijst de Skip link naar een `main` met `id="site-content"`, dit onderdeel bevindt zich direct onder de Page Header (inclusief navigatie).
  - Op een [detailpagina](https://www.denhaag.nl/nl/parkeren/bezoek-voor-parkeren-aan-en-afmelden) en [zoekpagina](https://www.denhaag.nl/nl/?s=ado) verwijst de Skip link naar een `main` met `id="site-content"`, de Breadcrumbs zijn het eerste onderdeel van main.
- De Skip link staat als losse link in `header` vlak voor het logo.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

- Geen specifiek component gevonden in de documentatie website of Storybook. Wel is er [documentatie over Skip links](https://dso-toolkit.nl/62.8.4/components/navigatie/anchor#skiplinks-en-jumplinks-wanneer-te-gebruiken) waarin wordt beschreven wanneer en hoe te gebruiken. Daarbij wordt ook een voorzet gedaan voor labels.
- Skip link wordt toegepast op https://aandeslagmetdeomgevingswet.nl/

---

### Screenshots / Varianten

**Website**
![Scherm­afbeelding 2024-05-15 om 09 50 45](https://github.com/nl-design-system/backlog/assets/545207/b1bb9775-2d98-4910-8a9f-03206e103ae2)

---

### Notities

Website
- Richtlijnen beschikbaar documentatie website.
- Positie: Bovenaan, over de content heen.
- Visual design: Gecentreerde donkere tekst op lichtgrijs vlak die over de gehele breedte uitvult. 
- Label: 'Ga naar de inhoud'
- Focus:
  - Op [homepage](https://aandeslagmetdeomgevingswet.nl/) verwijst de Skip link naar de eerste, lege, `div` binnen een `main` met `id="content-start"` en `tabindex="-1"`, dit onderdeel bevindt zich direct onder de Page Header (inclusief navigatie).
  - Op een [detailpagina](https://aandeslagmetdeomgevingswet.nl/implementatie/digitaal-stelsel/transitie-ruimtelijkeplannen-nl/) en [zoekpagina](https://aandeslagmetdeomgevingswet.nl/zoeken/?zoeken_term=omgeving) verwijst de Skip link naar de eerste, lege,  `div` binnen een `main` met `id="content-start"` en `tabindex="-1"`, dit onderdeel bevindt zich direct onder de Page Header (inclusief navigatie als breadcrumb).
- De Skip link staat binnen een `div` als losse link in `header`.

**Comment by jeffreylauwers** [comment]
## DUO

- Geen specifiek component gevonden in de documentatie website. Wel is er [documentatie over Skip links](https://uno.dfront.rijkscloud.nl/#/richtlijnen/accessibility) waarin wordt beschreven wanneer en hoe te gebruiken. Daarbij wordt ook een voorzet gedaan voor code en label.
- Skip link wordt toegepast op https://duo.nl/particulier/

---

### Screenshots / Varianten

**Website**
![Scherm­afbeelding 2024-05-15 om 10 36 51](https://github.com/nl-design-system/backlog/assets/545207/ddc9f272-d623-4a33-8bf3-44598482b0bf)

---

### Notities

Website
- Richtlijnen beschikbaar documentatie website.
- Positie: Linksboven, drukt content naar beneden.
- Visual design: als algemene link, breedte op basis van tekst.
- Label: 'Ga direct naar inhoud'
- Focus:
  - Op [homepage](https://duo.nl/particulier/) verwijst de Skip link naar een `main` met `id="content"` inclusief `class="skiplink-target main"` en `tabindex="-1"`, dit onderdeel bevindt zich direct onder de Page Header (zonder navigatie).
  - Op een [detailpagina](https://duo.nl/particulier/studiefinanciering/) en [zoekpagina](https://duo.nl/particulier/system/zoekresultaat.jsp?q=studie) verwijst de Skip link naar een `main` met `id="content"` inclusief `class="skiplink-target main"` en `tabindex="-1"`, dit onderdeel bevindt zich direct onder de Page Header (met navigatie).
- De Skip link staat als losse link in een `div` vlak voor de navigatie naar Particulier, Zakelijk, etc.

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

- [Patternlab](https://patternlab.lenteveld.nl/patterns/02-organisms-001-Generic-Page-Components-10-Skip-Navigation-OpenGemeentenSite-SkipNavigation/02-organisms-001-Generic-Page-Components-10-Skip-Navigation-OpenGemeentenSite-SkipNavigation.html) (tab toets gebruiken)
- Skip link wordt toegepast op https://www.lenteveld.nl/

---

### Screenshots / Varianten

**Website**
![Scherm­afbeelding 2024-05-15 om 11 14 36](https://github.com/nl-design-system/backlog/assets/545207/00c88f9d-9fed-416c-af1d-8d98eb479938)

---

### Notities

Website
- Positie: Linksboven, over de content heen.
- Visual design: Witte tekst op donkerblauw vlak, breedte op basis van tekst.
- Labels:
  - 'Naar hoofdinhoud'
  - 'Naar hoofdnavigatiemenu'
  - 'Naar footer'
  - 'Naar zoekveld'
- Focus:
  - Op [homepage](https://www.lenteveld.nl/) verwijzen de Skip links naar verschillende onderdelen:
    - 'Naar hoofdinhoud' verwijst **direct** naar `main` al heeft deze ook `id="content"`, dit onderdeel bevindt zich direct onder de Page header (inclusief navigatie).
    - 'Naar hoofdnavigatiemenu' verwijst naar een `nav` met `id="page-header__menu"`, dit onderdeel bevindt zich in de `header`.
    - 'Naar footer' verwijst naar een `footer` met `id="page-footer"`, dit onderdeel bevindt zich onder de Page content.
    - 'Naar zoekveld' verwijst naar een `input` met `id="page-header__search__input"`, dit onderdeel bevindt zich in de `header`.
  - Op een [detailpagina](https://www.lenteveld.nl/aanvragen-en-regelen/paspoort-aanvragen) en [zoekpagina](https://www.lenteveld.nl/zoeken?q=lente) verwijzen de Skip links naar verschillende onderdelen:
    - 'Naar hoofdinhoud' verwijst **direct** naar `main` al heeft deze ook `id="content"`, dit onderdeel bevindt zich direct onder de Page header (inclusief navigatie). Breadcrumbs (en eventuele Alert) vallen binnen deze `main`.
    - 'Naar hoofdnavigatiemenu' verwijst naar een `nav` met `id="page-header__menu"`, dit onderdeel bevindt zich in de `header`.
    - 'Naar footer' verwijst naar een `footer` met `id="page-footer"`, dit onderdeel bevindt zich onder de Page content.
    - 'Naar zoekveld' verwijst naar een `input` met `id="page-header__search__input"`, dit onderdeel bevindt zich in de `header`. Dus niet naar het 2e zoekveld binnen `main`.
- De Skip links staan als 4 losse links in een `nav` met `aria-label="skipnav menu"`. Deze `nav` met Skip links bevinden zich in een `div` vooraf aan een `div` met `class="page-container` waarbinnen de `header`, `main`, `footer` staan.

**Comment by jeffreylauwers** [comment]
## Rotterdam

Geen specifiek component gevonden in Figma of Storybook maar de Skip link wordt toegepast op https://www.rotterdam.nl/

---

### Screenshots / Varianten

**Website**
![Scherm­afbeelding 2024-05-15 om 11 46 58](https://github.com/nl-design-system/backlog/assets/545207/f943aabd-15e4-4430-9f60-fc9250c88bed)


---

### Notities

Website 
- Positie: Bovenaan in het midden, over de content heen.
- Visual design: Witte tekst op groen vlak, breedte op basis van tekst.
- Schuift van bovanaf in beeld.
- Label: 'Ga naar de hoofdinhoud'
- Focus:
  - Op [homepage](https://www.rotterdam.nl/) verwijst de Skip link naar een `main` met `id="top-of-content"`, dit onderdeel bevindt zich direct onder de Page Header (inclusief navigatie).
  - Datzelfde geldt voor een [detailpagina](https://www.rotterdam.nl/grofvuil-wegbrengen-of-laten-ophalen) waarbij de breadcrumbs binnen de `main` vallen.
  - En ook voor een [zoekpagina](https://www.rotterdam.nl/zoeken?zoekopdracht=feyenoord) waarbij de complete inhoud van de pagina zich binnen deze `main` bevindt.
- De Skip link staat als losse link in een `div` vlak voor een `div` waar de complete Page header (dus inclusief navigatie element met 'zoeken', 'contact', etc) mee is opgemaakt.

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-skip-link--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/skip-link)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=1862-18990&t=LPLYlQjrMrq2csFN-4)
- Skip link wordt toegepast op https://www.rvo.nl/

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-05-15 om 12 29 53](https://github.com/nl-design-system/backlog/assets/545207/f6b788b2-4f9b-4926-b0a7-7d37847cdd75)

**Figma**
![Scherm­afbeelding 2024-05-15 om 12 30 13](https://github.com/nl-design-system/backlog/assets/545207/2ffeb65e-55e6-4f1e-9048-230d8fcd6ba6)

**Website**
![Scherm­afbeelding 2024-05-15 om 12 31 06](https://github.com/nl-design-system/backlog/assets/545207/9d3fc8fc-c2a3-4a42-95e7-dd5e874092a5)

---

### Notities

Website 
- Positie: bovenaan, drukt content naar beneden.
- Visual design: Gecentreerde witte tekst op donkerblauw vlak die over de gehele breedte uitvult.
- Label: 'Direct naar de content'
- Focus:
  - Op [homepage](https://www.rvo.nl/) verwijst de Skip link naar een `div` met `id="main-content"` inclusief `role="main"` en `tabindex="0"`, dit onderdeel bevindt zich direct onder de Page Header (inclusief navigatie).
  - Datzelfde geldt voor een [detailpagina](https://www.rvo.nl/onderwerpen/vissersvaartuigen) waarbij de breadcrumbs binnen deze `div` vallen. Wat opvalt is dat de url van de Skip link de gehele url (`href="https://www.rvo.nl/onderwerpen/vissersvaartuigen#main-content"`) benoemd in plaats van enkel een verwijzing naar een`id`. 
  - Datzelfde geldt voor een [zoekpagina](https://www.rvo.nl/zoeken?) waarbij de complete inhoud van de pagina zich binnen deze `div` bevindt. Ook op deze pagina bevat de Skip link de gehele url (`href="https://www.rvo.nl/search?query=zaken%20doen%20in%20nederland%20do#main-content"`).
- De Skip link staat als losse link vlak voor een `div` waar de complete Page header mee is opgemaakt.

De opmaak van de website wijkt af van de code in Storybook waar wel direct naar `#main` wordt gelinkt.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-skip-link--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/skip-link)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=2217-10123&t=9ckJ5m5KSSOSTrjT-4)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-05-15 om 12 52 39](https://github.com/nl-design-system/backlog/assets/545207/de850ba4-a072-4f16-b676-95dc215bba5d)

**Figma**
![Scherm­afbeelding 2024-05-15 om 12 52 48](https://github.com/nl-design-system/backlog/assets/545207/6f5c9df6-52c7-450a-ae60-d8800f78ea74)

**Website**
![Scherm­afbeelding 2024-05-15 om 12 55 41](https://github.com/nl-design-system/backlog/assets/545207/ca8db7cc-0a4e-42be-95be-5e7e8f345d19)

---

### Notities

[Uitgebreide documentatie in Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-skip-link--docs).

Website
- Positie: Linksboven maar rechts van logo, over de content heen.
- Visual design: Rode tekst op wit vlak, breedte op basis van tekst.
- Labels:
  - 'Naar inhoud'
  - 'Naar menu'
  - 'Naar zoekveld'
- Focus:
  - Op de [homepage](https://www.utrecht.nl/) verwijzen de Skip links naar verschillende onderdelen:
    - 'Naar inhoud' verwijst naar `article` met `id="page-content"`, dit onderdeel bevindt zich direct onder de Page header (inclusief navigatie).
    - 'Naar menu' verwijst naar een `nav` met `id="menu"` en `tabindex="-1"`, dit onderdeel bevindt zich in de `header`.
    - 'Naar zoekveld' verwijst naar een `input` met `id="searchHomepage"`, dit onderdeel bevindt zich in de `header`.
  - Op een [detailpagina](https://www.utrecht.nl/wonen-en-leven/wonen/) verwijzen de Skip links naar verschillende onderdelen:
    - 'Naar inhoud' verwijst naar `article` met `id="page-content"`, dit onderdeel bevindt zich direct onder de Breadcrumbs.
    - 'Naar menu' verwijst naar een `nav` met `id="menu"` en `tabindex="-1"`, dit onderdeel bevindt zich in de `header`.
    - 'Naar zoekveld' verwijst naar een `input` met `id="searchHomepage"`, dit onderdeel bevindt zich in de `header`.
  - Op een [zoekpagina](https://www.utrecht.nl/over-deze-website/zoektips/zoeken/?q=utrecht) verwijzen de Skip links naar verschillende onderdelen:
    - 'Naar inhoud' verwijst naar `article` met `id="page-content"`, dit onderdeel bevindt zich direct onder de Breadcrumbs.
    - 'Naar menu' verwijst naar een `nav` met `id="menu"` en `tabindex="-1"`, dit onderdeel bevindt zich in de `header`.
    - 'Naar zoekveld' verwijst naar een `input` met `id="searchHomepage"`, dit onderdeel bevindt zich in de `header`.
- De Skip links staan als 3 losse links in een `div` binnen `header` vlak voor het logo. Deze links bevatten naast `title` ook `accesskey`.

**Comment by Ollie-nl** [comment]
We hebben een andere variant op ons loket staan: 
Deze komt uit het Design System (is een nieuwere omgeving)

![image](https://github.com/nl-design-system/backlog/assets/48805670/a91f0a5e-4cfb-4b6f-ae9c-ed1f5832d243)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Skip link op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=Skip+link)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Skip link op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=component%3A+skip+link)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Candidate ✨

Help je mee hem door de Hall of Fame stappen te krijgen?
[Skip Link op het Hall of Fame bord](https://github.com/orgs/nl-design-system/projects/30/views/1?filterQuery=skip+link)
